documentation/API-design-guidelines.md

```diff
@@ -1287,44 +1290,54 @@ response:
 
 
 ### 12.2 Event notification
 
 The event notification endpoint is used by the API server to notify the API consumer that an event occured.
 
-Note: The notification is the message posted on listener side. We describe the notification message in the CAMARA OAS but it could confusing because this endpoint should be implemented on the business API consumer side. This notice should be explicited mentioned in all CAMARA API documentation featuring notifications.
+CAMARA event notification leverages **[CloudEvents](https://cloudevents.io/)** as it is a vendor-neutral specification for defining the format of event data. A generic neutral CloudEvent notification swagger is available in commonalities/artifact directory (notification-as-cloud-event.yaml).
+
+Note: The notification is the message posted on listener side. We describe the notification(s) in the CAMARA API using the `callbacks`. From API consumer it could confusing because this endpoint must be implemented on the business API consumer side. This notice should be explicited mentioned in all CAMARA API documentation featuring notifications.
 
-Only Operation POST is provided for `eventNotifications` and the expected response code is `204`.
+Only Operation POST is provided for `eventNotifications` and the expected response code is `204`. 
+The url for this `POST` operation must be specified in the swagger as `{$request.body#/webhook/notificationUrl}`. 
 
-For consistence among CAMARA APIs the uniform `eventNotifications` model must be used:
+For consistence among CAMARA APIs the uniform CloudEvents model must be used with following rules:
 
 | name | type | attribute description | cardinality |
 | ----- |	-----  |	 -----  |  -----  | 
-| eventSubscriptionId | string | subscription identifier - could be valued for Resource-based subscription | optional |
-| event | object | event structure - see next table | mandatory |
+| id | string | identifier of this event, that must be unique in the source context. | mandatory |
+| source | string - URI | identifies the context in which an event happened in the specific Provider Implementation. Often this will include information such as the type of the event source, the organization publishing the event or the process that produced the event. The exact syntax and semantics behind the data encoded in the URI is defined by the event producer. | mandatory |
+| type | string | a value describing the type of event related to the originating occurrence. For consistency accross API we mandate following pattern: `org.camaraproject.<api-name>.<EVENT_NAME` (for exemple org.camaraproject.device-status.ROAMING_CHANGE_COUNTRY ) | mandatory |
```

Comment:
As this attribute contains a value describing the type of event related to the originating occurrence. The type of the event could be prefixed before the API name
org.camaraproject.<type/category>.<api-name>.<EVENT_NAME>

for eg.
org.camaraproject.network.qod.<EVENT_NAME>
org.camaraproject.device.device-status.<EVENT_NAME>
org.camaraproject.device.verify-location.<EVENT_NAME>

This could later align well our agenda number 4? WDYT?

Reply:
@shilpa-padgaonkar 

Point 1: I do not think so. I think point raised by @PedroDiez in point 4 is about having or not subscription endpoint in same yaml than the direct response endpoint. 

Point 2: I need to understand - We provide in the yaml the definition of the notification (via the `callback`) - So for example in the Device Status API v2 you'll find the notification about Roaming status change.  With your proposal we introduce the API version in the type like `"type:org.camara.api.device-status.v2.RoamingStatusEvent`. I'm correct ?

Reply:
For Point 1, I don't see the necessity for the <type> part. The format is to be make event types universally unique, and under the scope of "org.camaraproject", I think that <api-name> is going to be unique. With that 2 different CAMARA APIs could define same <EVENT_NAME> but it wouldn't be a problem.

Regarding event versioning, if would be useful if we set that schema for internal data is linked to the event type, and in order to change the schema for data we have to update the event type, or at least when a backwards incompatible change happen.

Reply:
For point 1: IF we simply assume the scope (of point 4)to be whether subscription yaml should be a separate file or not, you are correct, this is not needed. I am thinking more from a generic resource based subscription view, and in addition bringing the API families and later events under some structure. But may be this topic can also be asked in TSC & for now can be ignored here.

For point2: IF we consider resource based subscriptions, I don't believe that every event (in future) will have its source under some restful API like we have in the current cases. When we come to this point, it would be necessary to maintain the event versions to have visibility on the change in schema. 
Even though, this might be redundant as you have shown in your example for the current APIs in question, it will help keep the format consistent for all cases.